src/dump.c

```diff
@@ -570,7 +570,7 @@ static void jl_serialize_value_(jl_serializer_state *s, jl_value_t *v, int as_li
         write_uint8(s->s, TAG_METHOD);
         jl_method_t *m = (jl_method_t*)v;
         int internal = 1;
-        internal = module_in_worklist(m->module);
+        internal = m->is_for_opaque_closure || module_in_worklist(m->module);
```

Comment:
```suggestion
        if (m->is_for_opaque_closure)
             jl_error("unimplemented: code contained opaque closures");
        internal = module_in_worklist(m->module);
```

Reply:
I think the only thing that can reference this Method currently is the CodeInfo that creates it. This seems ok in that context. Do you see an issue?

Reply:
inlining can mess with that, but I guess this makes sense in contextual scope now